src/genophenocorr/preprocessing/_protein.py

```diff
@@ -35,7 +35,10 @@ def get_annotations(self, protein_id: str) -> typing.Optional[ProteinMetadata]:
         fpath = self._create_file_name(protein_id)
         if os.path.isfile(fpath):
             with open(fpath, 'rb') as fh:
-                return pickle.load(fh)
+                result_list = pickle.load(fh)
```

Comment:
I think you may be fixing the wrong thing here.

The change of `ProteinMetadataService` in `cd6dfa295e07d878d9b71e7ff1498056a7b51a3c` results in returning (and pickling) of a singluar instance, and not a list. Therefore, the pickle files are obsolete and must be updated. The code of this function should not be changed.

In other words, previously, we may have been serializing a list of `ProteinMeatadata` instead of a singular instance. Therefore, you have weird old pickle files, and it seems to me you want to fix them by updating the code, while the solution is to wipe the old pickle files.

Reply:
Yah, I reran this after deleting the .genophenocorr_cache/ folder and it ran fine before the changes. I didn't realize how old my files were. this is fixed.